libbeat/ebpf/types.go

```diff
@@ -0,0 +1,29 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ebpf
+
+import (
+	"github.com/elastic/ebpfevents"
+)
+
+type EventMask uint64
+
+type Watcher interface {
+	Subscribe(string, EventMask) <-chan ebpfevents.Record
+	Unsubscribe(string)
+}
```

Comment:
Documentation on this type would be nice (minimally the names of the params), but some indication of the expected semantics will help someone late at night fixing something.

Is there a reason we have an interface for this? Can we not work with a concrete type? AFAICS there is only one implementation.

Reply:
Since this watcher code is currently in two different PRs, and has been diverging, I've created a 3rd PR with only the watcher: https://github.com/elastic/beats/pull/37833

I've made all the suggested changes to the watcher in that PR, and will use that code once it's merged.